infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryService.kt

```diff
@@ -578,8 +578,9 @@ class GeometryService @Autowired constructor(
             // Pairs of (track meter, segment index). Ordinary ticks don't need segment indices because they clearly
             // hit a specific segment; but points on different sides of a segment boundary are often the exact same
             // point, but potentially have different heights (or more often null/not-null heights).
-            val allTicks = ((0..lastPoint step tickLength).map { distance -> distance.toBigDecimal() to null } +
-                    (planBoundaryAddressesByKm[kmNumber] ?: listOf())).sortedBy { (trackMeterInKm) -> trackMeterInKm }
+            val allTicks = ((planBoundaryAddressesByKm[kmNumber] ?: listOf()) +
+                    ((0..lastPoint step tickLength).map { distance -> distance.toBigDecimal() to null }))
+                .sortedBy { (trackMeterInKm) -> trackMeterInKm }
```

Comment:
Tehdäänkö jossain myöhemminmitään sillä, että `allTicks` on `Pair`? Tässä `second` mäpätään aina `null`:iksi, eikä tuossa myöhemminkään tarkastella mitään muuta kuin `first`:iä. Näkisin itse selkeämmäksi että tässä näitä käsiteltäisiin vain ei-pareina ja sitten jossain mahdollisesa myöhemmässä käyttöpaikassa ne mäpättäisiin pariksi jos on tarve

Reply:
Normaalitikit mäpätään nulleiksi joo, mutta `planBoundaryAddressesByKm`-puolelta tulevilla on segmentti-indeksitieto.